Client/Frontend/LoginManagement/LoginListDataSourceHelper.swift

```diff
@@ -62,7 +62,8 @@ class LoginListDataSourceHelper {
         var sections = [Character: [LoginRecord]]()
         var titleSet = Set<Character>()
 
-        DispatchQueue.global(qos: .userInteractive).async { [weak self] in
+        // Due to thread priority mismatches, we need to force this onto main to avoid app crashing. 
+        ensureMainThread { [weak self] in
```

Comment:
I'm not sure I understand why this needs to be the main thread. Does setting the qos to default also crash?

Looking at the code in the function I'm not sure why this is being dispatched to a new thread at all. Can we remove this line and leave it run on whatever thread initiated the request?

Reply:
Yeah let me get back to you on that.

Reply:
Okay so doing either of those leads to a crash. Removing Dispatches onto main in other areas leads to a crash in a `cellForRowAt` inside `LoginDataSource` when a user tries to filter logins / passwords. 

It seems like there's a mix of Dispatches to threads of varying qos in this area that might take more time to track and resolve. But there are comments here and there saying [completion needs to run on main thread](https://github.com/mozilla-mobile/firefox-ios/blob/5a0ed3c1967400758241e1d410853d96199aad6d/Client/Frontend/LoginManagement/LoginListViewModel.swift#L133).

Reply:
After discussing over chat, we decided to move the ensureMain to a level higher to capture the most cases of timing issues in this context.